src/views/disciplinas/components/SaveTreatmentForm.js

```diff
@@ -3,72 +3,35 @@ import { CButton, CFormLabel, CFormInput, CSpinner } from '@coreui/react'
 import { useHistory } from 'react-router-dom'
 import api from 'src/services/api'
 import Swal from 'sweetalert2'
-// import { fetchTreatments } from '../services/useFetchTreatment'
+import { useForm } from 'react-hook-form'
+import { useSubject } from '../hooks/useSubject'
 
 export const SaveTreatmentForm = () => {
-  // const [treatments, setTreatments] = useState([])
-  const [treatmentId, setTreatmentId] = useState('')
-  const [price, setPrice] = useState('')
-  const [loading, setLoading] = useState(false)
-  const history = useHistory()
-
-  // useEffect(() => {
-  //   fetchTreatments()
-  //     .then((result) => {
-  //       setTreatments(result)
-  //     })
-  //     .catch((error) => {
-  //       console.log(error?.response)
-  //     })
-  // }, [])
-
-  const handleSubmit = async () => {
-    console.log(setTreatmentId)
-    setLoading(true)
-    try {
-      const salonId = JSON.parse(String(localStorage.getItem('user-id')))
-      const sendData = {
-        salonId,
-        price: Number(price),
-        treatmentId,
-      }
-      await api.post(`/treatmentsalon`, sendData)
-      Swal.fire('Sucesso!', `Insreido com sucesso`, 'success')
-      setLoading(false)
-    } catch (error) {
-      console.log(error.response)
-      Swal.fire('Erro!', `${error?.response?.data.error}`, 'error')
-      setLoading(false)
-    }
-    history.go(0)
-  }
+  const { handlePostSubject, setLoading, loading } = useSubject()
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm()
 
   return (
-    <form>
+    <form onSubmit={handleSubmit(handlePostSubject)}>
       <div className="mb-3" width="100px">
         <CFormLabel htmlFor="disciplina">Disciplina</CFormLabel>
         <div style={{ display: 'flex', gap: 12 }}>
```

Comment:
Não sei como está sendo o padrão de estilização aqui neste projeto, mas penso que deves modificar aqui. Se é Tailwind, que faças da forma do Tailwind. Se for Reactstrap, que assim também o faças.